### ISSUE TITLE
Authorization of the API 

### ISSUE THREAD

**Issue by wtrocki** [oasis-open/odata-rapid#12]
Authorization for the CRUD is a very important factor. 
If `$filter` gives the ability to read very wide amounts of data it should be possible to restrict unauthorized users from reading some specific data.

Every GraphQL framework has this capability in the Core.
For example:
https://docs.aws.amazon.com/appsync/latest/devguide/security.html

With generic crud there are usually two different challenges:

- Owner based systems where generic crud capabilities should be 

## Questions

Rapid Pro defines some of the capabilities or even examples of the problem?
(As this was mentioned as a possible disadvantage of CRUD based systems comparing to declarative API)

## Example

Only owner can reference object and fetch it in filter
```graphql
type Task {
 id: ID! @RapidID
 name: String! @RapidSecurity(ownerOnly:true)
}
```

The only person with role

```graphql
type Task {
 id: ID! @RapidID
 name: String! @RapidSecurity(role: "admin")
}
```

**Comment by GEricson** [comment]
I agree that leaving this to implementations is the correct choice.  The primary issue is that authorization/authentication decisions should be able to be changed at run-time.  Defining them at schema-definition time would be limiting.  As an example, DMTF Redfish has implemented a comprehensive solution.  I'm sure @mikepizzo can point to others.
